tests/e2e-test.spec.js

```diff
@@ -43,17 +43,88 @@ describe('Main workflow Test Suite', () => {
+    const reportsTableResult = await client.query(`SELECT * FROM ${pgSchema}.reports`);
+    expect(reportsTableResult.rows.length).to.equal(reports().length);
+  });
+
+  it('should process document edits', async () => {
+    const report = reports()[0];
+    const contact = contacts()[0];
+    await editDoc({ ...report, edited: 1 });
+    await editDoc({ ...contact, edited: 1 });
+
+    await delay(6); // wait for CHT-Sync
+
+    const pgTableDataRecord = await client.query(`SELECT * from ${PGTABLE} where _id = $1`, [report._id]);
+    expect(pgTableDataRecord.rows[0].doc.edited).to.equal(1);
+
+    const pgTableContact = await client.query(`SELECT * from ${PGTABLE} where _id = $1`, [contact._id]);
+    expect(pgTableContact.rows[0].doc.edited).to.equal(1);
+
+    await delay(6); // wait for DBT
+
+    const modelReportResult = await client.query(`SELECT * FROM ${pgSchema}.reports where _id = $1`, [report._id]);
+    expect(modelReportResult.rows[0].doc.edited).to.equal(1);
+
+    const modelContactResult = await client.query(`SELECT * FROM ${pgSchema}.contacts where _id = $1`, [contact._id]);
+    expect(modelContactResult.rows[0].doc.edited).to.equal(1);
+
+    const contactsTableResult = await client.query(`SELECT * FROM ${pgSchema}.contacts`);
+    expect(contactsTableResult.rows.length).to.equal(contacts().length);
+
+    const reportsTableResult = await client.query(`SELECT * FROM ${pgSchema}.reports`);
+    expect(reportsTableResult.rows.length).to.equal(reports().length);
+  });
+
+  it('should process contact deletes', async () => {
+    const contact = contacts()[0];
+    await deleteDoc(contact);
+    await delay(6); // wait for CHT-Sync
+    const pgTableContact = await client.query(`SELECT * from ${PGTABLE} where _id = $1`, [contact._id]);
+    expect(pgTableContact.rows[0]._deleted).to.equal(true);
+    await delay(6); // wait for DBT
+    const modelContactResult = await client.query(`SELECT * FROM ${pgSchema}.contacts where _id = $1`, [contact._id]);
+    expect(modelContactResult.rows.length).to.equal(1);
+    expect(modelContactResult.rows[0]._deleted).to.equal(true);
+  });
+
+  it('should process person deletes', async () => {
+    const person = contacts().find(contact => contact.type === 'person' && !contact._deleted);
+
+    const preDelete = await client.query(`SELECT * FROM ${pgSchema}.persons where _id = $1`, [person._id]);
+    expect(preDelete.rows.length).to.equal(1);
+
+    await deleteDoc(person);
+    await delay(6); // wait for CHT-Sync
+    await delay(6); // wait for DBT
+    await delay(6); // wait for DBT
+
+    const modelContactResult = await client.query(`SELECT * FROM ${pgSchema}.contacts where _id = $1`, [person._id]);
+    expect(modelContactResult.rows[0]._deleted).to.equal(true);
+
+    const postDelete = await client.query(`SELECT * FROM ${pgSchema}.persons where _id = $1`, [person._id]);
+    expect(postDelete.rows[0]._deleted).to.equal(true);
```

Comment:
```suggestion
    expect(postDelete.rows.length).to.equal(1);
    expect(postDelete.rows[0]._deleted).to.equal(true);
```

Reply:
This is also addressed in this [PR](https://github.com/medic/cht-sync/pull/118)